server/routes/api/dashboard.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const router = express.Router();
+let {
+    Course
+} = require('../../dbSchemas/attendanceSchema');
+
+
+/**GETS all the courses registered by a student
+ * ==========================================
+ * Example api call body:
+ * {
+	"studentId": "5f984a44da9eb32ba01d31dd"
+    }
+ */
+router.get('/getCoursesByStudent', (req, res) => {
+    let studentID = req.body.studentId;
+
+    Course.find({
+        registered_students: {
+            $in: [studentID]
+        }
+    }, function (err, courses) {
+        if (err || courses == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+        res.status(200).send(courses);
+    });
+});
+
+/**GETS all the courses created by a professor
+ * ==========================================
+ * Example api call body:
+ * {
+	"studentId": "5f98967d62a8bb73a0f8c046"
+    }
+ */
+router.get('/getCoursesCreatedByProfessor', (req, res) => {
+    let professorID = req.body.professorId;
+
+    Course.find({
+        professor: professorID
+    }, function (err, courses) {
+        if (err || courses == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+        res.status(200).send(courses);
+    });
+});
+
+/** Registers a student to his course
+ *  We give a student ID
+ *  Returns course data (name, prof name, if it's mandatory)
+ */
+router.post('/registerForCourse', (req, res) => {
+    let studentID = req.body.studentID;
+    let courseID = req.body.courseID;
+    console.log("register for course");
+
+    Course.findOne({
+        _id: courseID
+    }, function (err, course) {
+        if (err || course == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+
+        // push new student into array
+        course.registered_students.push(studentID);
+
+        let courseName = course.name;
+        let prof = course.professor;
+        let mandatory = course.always_mandatory;
+
+        course.save(err => {
+            if (err) {
+                console.log(err);
+                return res.status(500).send(err);
+            }
+            // return course information
+            return res.status(200).json({
```

Comment:
Personally, I think you should just return the whole course just to make the api endpoint more flexible and look cleaner. But you did return what the story card asked for.

Reply:
yeah, that's definitely easier. I assume the frontend won't mind sorting that. There could be some security concerns with that, though.

Reply:
There isn't really any personal information directly in the course itself so it should be fine for security (it only has the prof's id and the student's ids, not their actual data)

Reply:
Ok, sounds good then!